### PR TITLE
Improving error message for service not in default namespace

### LINKED ISSUE #5271: Improve error message for service not in default namespace.
1- create a NodePort service in not default namespace:

```
$ kubectlget svc -A
NAMESPACE     NAME                    TYPE        CLUSTER-IP       EXTERNAL-IP   PORT(S)                  AGE
default       kubernetes              ClusterIP   10.96.0.1        <none>        443/TCP                  15d
dgeek-v1      dgeek-helm-dgeek-helm   NodePort    10.103.186.237   <none>        80:32316/TCP             3m40s
```

2- try to open the service url:
```
$ minikube service dgeek-helm-dgeek-helm

💣  Error opening service
❌  Error: [SERVICE_NOT_FOUND] Could not find finalized endpoint being pointed to by dgeek-helm-dgeek-helm: Temporary Error: Error getting service dgeek-helm-dgeek-helm: services "dgeek-helm-dgeek-helm" not found
Temporary Error: Error getting service dgeek-helm-dgeek-helm: services "dgeek-helm-dgeek-helm" not found
Temporary Error: Error getting service dgeek-helm-dgeek-helm: services "dgeek-helm-dgeek-helm" not found
Temporary Error: Error getting service dgeek-helm-dgeek-helm: services "dgeek-helm-dgeek-helm" not found
💡  Suggestion: Please make sure the service you are looking for is deployed or is in the correct namespace.
⁉️   Related issues:
    ▪ https://github.com/kubernetes/minikube/issues/4599

😿  If the above advice does not help, please let us know: 
👉  https://github.com/kubernetes/minikube/issues/new/choose
```

the error message is big and suggestion gets over-shaddowed.

I suggest the error message say

`service X was not found in default namespace , please try with "minikube service X -n Y"`

### PR DESCRIPTION
**Pull request by rbrishabh**
Fixes #5271 

Hey @josedonizetti @medyagh 

Does this fix the issue?

I am very new with go-lang, so I might have gone wrong.
If I went wrong, please point me in the correct direction and I will be more than happy to look into this!

Thanks! =)

### CODE REVIEW

File: pkg/minikube/service/service.go [hunk 1]
```diff
@@ -266,7 +266,7 @@ func WaitAndMaybeOpenService(api libmachine.API, namespace string, service strin
 	chkSVC := func() error { return CheckService(namespace, service) }
 
 	if err := retry.Expo(chkSVC, time.Duration(interval)*time.Second, time.Duration(wait)*time.Second); err != nil {
-		return errors.Wrapf(err, "Could not find finalized endpoint being pointed to by %s", service)
+		return errors.Wrapf(err, "Service %s was not found in default namespace, please try again with 'minikube service %s -n %s'", service, service, namespace)
```
Comment (n0npax) [inline pkg/minikube/service/service.go]: Hi,
I'm not the core reviewer, but I believe it's not the best solution.
function `CheckService`
looks like:
```go
  212 // CheckService checks if a service is listening on a port.                                                             
  213 func CheckService(namespace string, service string) error {                                                             
  214         client, err := K8s.GetCoreClient()                                                                              
  215         if err != nil {                                                                                                 
  216                 return errors.Wrap(err, "Error getting kubernetes client")                                              
  217         }                                                                                                               
  218                                                                                                                         
  219         svc, err := client.Services(namespace).Get(service, meta.GetOptions{})                                          
  220         if err != nil {                                                                                                 
  221                 return &retry.RetriableError{                                                                           
  222                         Err: errors.Wrapf(err, "Error getting service %s", service),                                    
  223                 }                                                                                                       
  224         }                                                                                                               
  225         if len(svc.Spec.Ports) == 0 {                                                                                   
  226                 return fmt.Errorf("%s:%s has no ports", namespace, service)                                             
  227         }                                                                                                               
  228         glog.Infof("Found service: %+v", svc)                                                                           
  229         return nil                                                                                                      
  230 } 
```
what if error was returned due to `no ports`?
maybe before printing error about `default` namespace we should check if `namespace == "default"` and embed error msg from `CheckService`

Comment (rbrishabh) [reply pkg/minikube/service/service.go]: Hey @n0npax ! Thanks, I'll take another look at this today!

**Comment by tstromberg** [comment]
@minikube-bot OK to test

**Comment by sharifelgamal** [comment]
@minikube-bot OK to test

**Comment by medyagh** [comment]
@rbrishabh do you mind pasting before and after output of this PR ?

**Comment by rbrishabh** [comment]
I am not able to test this locally, unfortunately!

**Comment by medyagh** [comment]
@rbrishabh sorry this PR went stale. I just noticed we have a newer PR https://github.com/kubernetes/minikube/pull/5844 

sadly I have to close this one in favor of the newer one. I hope you still consider picking up another good-first issue

**Pull request closed without merging**